### PR TITLE
Add sqlite CRUD example

### PR DESCRIPTION
**Pull request by godsarmy**
This is a demo code almost same as https://www.youtube.com/watch?v=n2Z0pflkZQU but with a little bit improvement (such as better way to initialize sqlite file)

### CODE REVIEW

File: examples/sqlite/main.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    """Callback when a new user button is clicked"""
+    try:
+        cursor.execute(
+            "INSERT INTO users (name, age) VALUES (?, ?)", (name.value, age.value)
+        )
+        conn.commit()
+        ui.notify(f"success add new user {name.value}", color="blue")
+        name.value = ""
+        age.value = ""
+        list_alldata.clear()
```
Comment (rodja) [inline examples/sqlite/main.py:51]: You could take advantage of [ui.refreshable](https://nicegui.io/documentation#refreshable_ui_functions) to simplify the code.

File: examples/sqlite/main.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        get_all_data()
+    except Exception as e:
+        ui.notify(f"failed to add new user {name.value}: {e}", color="red")
+
+
+def get_all_data():
+    cursor.execute("SELECT * FROM users")
+    res = cursor.fetchall()
+    result = []
+    for row in res:
```
Comment (rodja) [inline examples/sqlite/main.py:61]: Maybe use list comprehension to simplify the construction of `result`?

File: examples/sqlite/main.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        editdialog.close()
+        list_alldata.clear()
+        get_all_data()
+    except Exception as e:
+        ui.notify(f"failed to edit user, {e}", color="red")
+
+
+def editdata(data):
+    """Callback when user edit button is clicked"""
+    get_id.text = data.default_slot.children[0].text
```
Comment (rodja) [inline examples/sqlite/main.py:102]: Grabbing the data from the UI is not super elegant. Maybe the example could benefit from an ORM like https://tortoise.github.io/?

**Review by rodja — CHANGES_REQUESTED** [review]
Thanks for the example. A small database example was really missing. I made some comments throughout the code where you may want to improve it a little. But overall I think it will greatly help other users! Thanks again.

**Comment by rodja** [comment]
@godsarmy I would really like this to be part of the 1.2.10 release. Do you think you can make the adaptations until tomorrow?

**Comment by godsarmy** [comment]
hi

I changed partial but i am super busy these days. I hope to complete all my
changes in this week.

Rodja Trappe ***@***.***> 于 2023年4月25日周二 11:06写道：

**Review by rodja — CHANGES_REQUESTED** [review]
I've made a few small comments in the code. Additionally your example should be listed on the website. Use the example_link function below the ["In-depth examples" section heading](https://github.com/zauberzeug/nicegui/blob/8a86d2064f8f4464f3819ac5c6763a2cb2d0e990/main.py#L242). The title should match the example folder name when [snake case converted](https://github.com/zauberzeug/nicegui/blob/8a86d2064f8f4464f3819ac5c6763a2cb2d0e990/website/style.py#L31).

I'm also still in favour for using [tortoise ORM](https://tortoise.github.io/) in this example. The db access is quite low level and uses blocking io which is suboptimal. But we can do it once this PR is integrated.

**Comment by rodja** [comment]
One other thing: do you think we can name the example "database"?

**Comment by godsarmy** [comment]
> One other thing: do you think we can name the example "database"?

I still prefer to name it sqlite while i would like to provide another example in the future to use ORM to operate db named as  "database"

**Comment by rodja** [comment]
> I still prefer to name it sqlite while i would like to provide another example in the future to use ORM to operate db named as "database"

Why would we need two examples? I would expect the ORM example to get simpler and simultaneously more production-ready than the  raw sqlite example...

File: examples/sqlite/db/users.sql
Comment (rodja) [inline examples/sqlite/db/users.sql:1]: Is this file needed?

**Comment by falkoschindler** [comment]
I just reviewed the code and committed several small improvements:

- remove the unused db file
- add a .gitignore
- improve type annotation
- remove some expendable whitespace, comments, docstrings and exception handlers
- use more consistent function and variable names
- use `ui.number` consistently`
- fix initial `conn.commit()` not being called
- use `on_click` instead of `.on('click')`
- use `color` attribute instead of background color classes
- use notification type instead of color
- use global `dialog_id` variable instead of `get_id` label (still not perfect, but ok for this example)
- avoid using UI elements for passing around data, use `user` dictionary instead

Removing exception handling is certainly disputable. But I chose to cut some corners to shorten the example and to focus on the essence.

Using an ORM remains an open task.

**Review by rodja — APPROVED** [review]

**Comment by godsarmy** [comment]
> I just reviewed the code and committed several small improvements:
> 
> * remove the unused db file
> * add a .gitignore
> * improve type annotation
> * remove some expendable whitespace, comments, docstrings and exception handlers
> * use more consistent function and variable names
> * use `ui.number` consistently
> * fix initial `conn.commit()` not being called
> * use `on_click` instead of `.on('click')`
> * use `color` attribute instead of background color classes
> * use notification type instead of color
> * use global `dialog_id` variable instead of `get_id` label (still not perfect, but ok for this example)
> * avoid using UI elements for passing around data, use `user` dictionary instead
> 
> Removing exception handling is certainly disputable. But I chose to cut some corners to shorten the example and to focus on the essence.
> 
> Using an ORM remains an open task.

HI Thank you for the change. I think the change makes sense. As a new new user for nicegui, i get to know the best practice. I will submit another PR to convert the example modeled in ORM.

**Comment by rodja** [comment]
I finally reworked the [sqlite database example](https://github.com/zauberzeug/nicegui/blob/main/examples/sqlite_database/main.py) to use Tortoise ORM.

**Pull request merged**